spring-test/src/main/java/org/springframework/mock/web/MockHttpServletRequest.java

```diff
@@ -921,6 +940,21 @@ else if (value != null) {
 		}
 	}
 
+	private long parseDateHeader(String name, String value) {
+		for (String dateFormat : DATE_FORMATS) {
+			SimpleDateFormat simpleDateFormat = new SimpleDateFormat(dateFormat, Locale.US);
+			simpleDateFormat.setTimeZone(GMT);
+			try {
+				return simpleDateFormat.parse(value).getTime();
+			}
+			catch (ParseException ex) {
+				// ignore
+			}
+		}
+		throw new IllegalArgumentException("Cannot parse date value \"" + value +
```

Comment:
AFAIK, in Spring we typically enclose _quoted_ values within exception messages in single quotes.